code/modules/projectiles/guns/energy.dm

```diff
@@ -76,12 +78,28 @@
 		power_supply = new cell_type(src)
 	else
 		power_supply = new /obj/item/cell/device/variable(src, max_shots*charge_cost)
+	if(charge_module)
+		self_recharge = TRUE
+		use_external_power = FALSE
+		recharge_time = charge_module.charge_rate
 	update_maptext()
 
 /obj/item/gun/energy/Destroy()
 	QDEL_NULL(power_supply)
+	if(istype(charge_module))
+		qdel(charge_module)
 	return ..()
 
+/obj/item/gun/energy/attackby(obj/item/I, mob/user)
+	. = ..()
+	if(charge_module && I.iswirecutter())
+		to_chat(user, SPAN_NOTICE("You snip the wires connecting [charge_module] to \the [src]'s battery'."))
+		charge_module.forceMove(get_turf(src))
+		user.put_in_hands(charge_module)
+		charge_module = null
+		self_recharge = FALSE
```

Comment:
You need to call a proc to the module itself to drop the reference to the gun once disconnected, see below

Reply:
addressed